tools/DeleteModules.ps1

```diff
@@ -0,0 +1,208 @@
+        if ($scope -eq "AzureRMAndDependencies") {
+            $query = "AzureRM"
+        } elseif ($scope -eq "AzureAndDependencies") {
+            $query = "Azure"
+        } elseif ($scope -eq "NetCoreModules") {
+            $query = "AzureRM.Netcore"
+        } elseif ($scope -eq "AzureStackAndDependencies") {
+            $query = "AzureStack"
+        }
+
+        $azureRmAllVersions = Find-Module -Name $query -Repository PSGallery -AllVersions
+        $targets += $azureRmAllVersions[0]
+        $currentDependencies = $azureRmAllVersions[0].Dependencies
+        $previousDependencies = $azureRmAllVersions[1].Dependencies
+        $currentDependencies | ForEach-Object {
+            $CDModule = $_
+            $versionChanged = $true
+            $previousDependencies | ForEach-Object {
+                if (($_.Name -eq $CDModule.Name) -and (($_.RequiredVersion -eq $CDModule.RequiredVersion) -and ($_.MinimumVersion -eq $CDModule.MinimumVersion)))
+                {
+                    $versionChanged = $false
+                }
+            }
+
+            if ($versionChanged)
+            {
+                $targets += $CDModule
+            }
+        }
+
+        return $targets
+    }
+}
+
+function Get-DependentModules
+{
+    param(
+        [string]$repoName,
+        [string]$moduleName,
+        [string]$moduleVersion,
+        $allModules,
+        $azureModules
+    )
+
+    $dependencies = @()
+    $azureModules | ForEach-Object {
+        $targetName = $_.Name
+        $moduleNotBeingDeleted = $true
+        $allModules | ForEach-Object {
+            if ($_.Name -eq $targetName)
+            {
+                $moduleNotBeingDeleted = $false
+            }
+        }
+
+        $_.Dependencies | ForEach-Object {
+            $dependencyName = $_.Name
+
+            if ($moduleNotBeingDeleted -and ($moduleName -eq $dependencyName))
+            {
+                $allTargetVersions = Find-Module -Name $targetName -Repository $repoName -AllVersions
+                $allTargetVersions | ForEach-Object {
+                    $targetVersion = $_.Version
+                    $_.Dependencies | ForEach-Object {
+                        if (($moduleName -eq $_.Name) -and (($_.MinimumVersion -eq $moduleVersion) -or ($_.RequiredVersion -eq $moduleVersion)))
+                        {
+                            $dependencies += "`"$targetName $targetVersion`","
+                        }
+                    }
+                }
+            }
+        }
+    }
+    if ($dependencies.Count -ne 0)
+    {
+        Write-Warning "$moduleName $moduleVersion is a dependency for $dependencies the module(s) will have an orphaned dependency."
+    }
+}
+
+function Get-ApiKey
+{
+	param([string]$repoName)
+    
+    $vaultKey="PSTestGalleryApiKey"
+    if ($repoName -eq "PSGallery")
+    {
+        $vaultKey = "PowerShellGalleryApiKey"
+    }
+    
+    $context = $null
+    try {
+        $context = Get-AzureRMContext
```

Comment:
You can use ```-ErrorAction Stop``` in this call to ensure that it throws whatever the setting is in the PowerShell session

Reply:
Done